AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m

```diff
@@ -134,16 +144,22 @@ - (void)enqueueItem:(id<MSLog>)item {
                                            code:kMSACConnectionSuspendedErrorCode
                                        userInfo:@{NSLocalizedDescriptionKey : kMSACConnectionSuspendedErrorDesc}];
       [self notifyFailureBeforeSendingForItem:item withError:error];
-      [self completedEnqueuingLog:item withInternalId:internalLogId withSuccess:NO];
+      [self enumerateDelegatesForSelector:@selector(channel:didCompleteEnqueueingLog:withInternalId:)
+                                withBlock:^(id<MSChannelDelegate> delegate) {
+                                  [delegate channel:self didCompleteEnqueueingLog:item withInternalId:internalLogId];
+                                }];
       return;
     }
 
     // Save the log first.
     MSLogDebug([MSAppCenter logTag], @"Saving log, type: %@.", item.type);
-    BOOL success = [self.storage saveLog:item withGroupId:self.configuration.groupId];
+    [self.storage saveLog:item withGroupId:self.configuration.groupId];
     self.itemsCount += 1;
-    [self completedEnqueuingLog:item withInternalId:internalLogId withSuccess:success];
-
+    [self enumerateDelegatesForSelector:@selector(channel:didCompleteEnqueueingLog:withInternalId:)
```

Comment:
Not a merge blocker but we should think about not repeating this three times in the method.

Reply:
Right